### PR TITLE
Use `pyproject.toml` for setuptools config

### PR DESCRIPTION
**Pull request by michael-the1**
Use `pyproject.toml` for setuptools config

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -2,6 +2,34 @@
 requires = ["setuptools>=42", "wheel", "setuptools_scm[toml]~=6.0"]
 build-backend = "setuptools.build_meta"
 
+[project]
+name = "diepvries"
+version = "0.9.2"
```
Comment (matthieucan) [inline pyproject.toml:7]: This should be a dynamic attribute, as the version is computed with git tags and `setuptools_scm`, right?

Comment (michael-the1) [reply pyproject.toml]: Yes, definitely. I also bumped the minimum setuptools and setuptools_scm version as per https://github.com/pypa/setuptools_scm?tab=readme-ov-file#pyprojecttoml-usage

**Review by matthieucan — COMMENTED** [review]
Could you also update the information in `CONTRIBUTING.md` which references `setup.py`?

File: pyproject.toml [hunk 2]
```diff
@@ -1,7 +1,35 @@
...
+    "License :: OSI Approved :: MIT License",
+    "Programming Language :: Python :: 3",
+    "Intended Audience :: Developers",
+    "Operating System :: OS Independent",
+]
+dependencies = ["snowflake-connector-python~=3.0"]
+requires-python = ">= 3.8"
+
+[project.optional-dependencies]
+test = ["pytest~=6.2", "tox"]
```
Comment (matthieucan) [inline pyproject.toml:23]: Should `tox` be here?

Comment (michael-the1) [reply pyproject.toml:23]: True, I guess it's a better fit in a `dev` group. I do think it's nice to have.

**Review by dlouseiro — APPROVED** [review]
Works like a charm for me! Good to go

File: pyproject.toml [hunk 3]
```diff
@@ -1,7 +1,36 @@
...
+test = ["pytest~=6.2"]
+dev = ["pytest~=6.2", "tox", "pre-commit"]
+
+[project.urls]
+Documentation = "https://diepvries.picnic.tech/"
+"Source Code" = "https://github.com/PicnicSupermarket/diepvries"
+"Issue Tracker" = "https://github.com/PicnicSupermarket/diepvries/issues"
+
+[tool.setuptools]
+package-dir = {where = "src"}
```
Comment (matthieucan) [inline pyproject.toml:32]: I don't think this is correct, it creates a `where/` directory in the wheel

**Review by matthieucan — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 3 above)
Comment (matthieucan) [reply pyproject.toml:32]: More precisely, it bundles `diepvries` under the top-level namespace `where`

Comment (michael-the1) [reply pyproject.toml]: Good catch. I found out that `setuptools>=61` automatically detects `src/` layouts: https://setuptools.pypa.io/en/latest/userguide/package_discovery.html#automatic-discovery So I removed these lines.

Comment (matthieucan) [reply pyproject.toml]: Nice, works perfectly!

**Review by matthieucan — APPROVED** [review]
🚀

**Pull request merged**